### ISSUE TITLE
error when I uncomment the destructor, confused

### ISSUE THREAD

**Issue by charlesw1234** [miloyip/rapidjson#545]
For the following code, it will works when I comment the destructor: ~container_t, but it can not pass the compilation if it is added. This confuse me, what is the reason?

https://github.com/charlesw1234/learning/blob/master/rapidjson/member.cpp

#include <vector>
#include "rapidjson/document.h"

class container_t {
public:
    container_t(void) {}
    //virtual ~container_t() {}

    inline void setup(const std::string &name, const rapidjson::Value &value,
                      rapidjson::Document::AllocatorType &allocator)
    {   _name = name; _value.CopyFrom(value, allocator); }
private:
    std::string _name;
    rapidjson::Value _value;
};

class containers_t: public std::vector<container_t> {};

int
main(void)
{
    containers_t containers;
    containers.resize(5);
    return 0;
}

**Comment by miloyip** [comment]
In C++03, this code should not compile because `Value` cannot be copy-constructed. In C++11 it is possible if the class or container only need to move-constructed `Value`.

Basically you cannot put `Value` or `Document` in member variable or inside STL containers. You will need to use pointer instead, if you really need to.

**Comment by charlesw1234** [comment]
Ok, I understand you. But why you do not provide a copy-constructed support for rapidjson::Value? for performance? and Allocator has to be provided?

**Comment by miloyip** [comment]
Yes. `Allocator` needs to be provided in order to clone a `Value`.
